server/gitrest/src/utils/definitions.ts

```diff
@@ -28,6 +29,18 @@ export interface IRepositoryManager {
     createTag(tagParams: git.ICreateTagParams): Promise<git.ITag>;
 }
 
+/**
+ * Subset of Node.js `fs/promises` API.
+ */
+export interface IFileSystemManager {
+    readFile: typeof fsPromises.readFile;
+    writeFile: typeof fsPromises.writeFile;
+    mkdir: typeof fsPromises.mkdir;
+    rmdir: typeof fsPromises.rmdir;
+    stat: typeof fsPromises.stat;
+    rm: typeof fsPromises.rm;
```

Comment:
We might also want to include APIs that isomorphic-git also uses:
* unlink
* readdir
* lstat

The following are optional in IsoGit in that the implementation can be dummy, but IsoGit expects a definition, even if dummy (e.g. throwing not implemented error):
* readlink 
* symlink
* chmod

List obtained from here: https://github.com/isomorphic-git/isomorphic-git/blob/main/docs/fs.md#using-the-promise-api-preferred